### PR TITLE
Auto: implement issue #602

### LINKED ISSUE #602: Add ticket_categorization SQLAlchemy ORM model and DB migration
Subtask of #45

Create a new ORM model `TicketCategorization` (or extend existing Ticket model) to store: type, priority, confidence, reasons (JSON), suggested_assignee_id, suggested_team, categorized_at, human_override (bool). Write an alembic migration. Do NOT wire it to any endpoint yet. Tests: add a unit test for the model and an integration test that creates and reads the record.

Dev-Plan: docs/dev-plan/30-tickets/0602-add-ticket-categorization-sqlalchemy-orm-model-and-db-migrat.md

### PR DESCRIPTION
**Pull request by yihua-zhuo**
Closes #602

Plan: `.plans/issue-602.md`

This PR was auto-implemented by Claude Code in three stages:

1. **Plan** — Claude wrote a detailed implementation plan to `.plans/issue-602.md`.
2. **Implement** — Claude executed the plan, ran `ruff check src/` + the unit and integration test suites, and iterated until all three were green.
3. **Self-review** — Claude reviewed its own diff against `codereview.md` (the project review checklist) and fixed any rule violations found, re-running the test suite if any changes were made.

PR-time CI (`Dev Agent CI`) re-verifies independently before this PR is mergeable.

Please review before merging.

## Summary by CodeRabbit

* **New Features**
  * Added ticket categorization persistence with support for category type, priority, confidence scores, reasoning details, and suggested assignments.
  * Categorization records now include human override tracking and automatic timestamps.

* **Tests**
  * Added comprehensive unit and integration test coverage for categorization data persistence and retrieval.

* **Chores**
  * Added implementation planning documentation for ticket categorization feature.

[Review Change Stack](https://app.coderabbit.ai/change-stack/yihua-zhuo/agent-job/pull/699?utm_source=github_walkthrough&utm_medium=github&utm_campaign=change_stack)

### CODE REVIEW

File: src/db/models/ticket_categorization.py [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+        ForeignKey("tickets.id", ondelete="CASCADE"),
+        nullable=False,
+    )
+    category_type: Mapped[str] = mapped_column(String(50), nullable=False)
+    priority: Mapped[str | None] = mapped_column(String(50), nullable=True)
+    confidence: Mapped[Decimal | None] = mapped_column(Numeric(5, 4), nullable=True)
+    reasons: Mapped[dict | None] = mapped_column(JSON, nullable=True)
+    suggested_assignee_id: Mapped[int | None] = mapped_column(Integer, nullable=True)
+    suggested_team: Mapped[str | None] = mapped_column(String(100), nullable=True)
+    human_override: Mapped[bool] = mapped_column(Boolean, nullable=False, default=False, server_default=text("false"))
```
Comment (yihua-zhuo) [inline src/db/models/ticket_categorization.py:31]: **WARNING** (suggestion)

**Issue:** human_override column defines both Python default=False and server_default=text('false'); the Python default is unused at INSERT time but may mislead readers about which default applies.

**Suggestion:** Remove default=False from the mapped_column call since the server_default controls the actual DB value: mapped_column(Boolean, nullable=False, server_default=text('false'))

File: tests/integration/test_ticket_categorization_integration.py [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+    async def test_create_and_fetch_categorization(self, db_schema, tenant_id, async_session):
+        """Full field round-trip: create a categorization and fetch it back."""
+        from models.ticket import TicketChannel, TicketPriority
+        from services.ticket_service import TicketService
+
+        svc = TicketService(async_session)
+        ticket = await svc.create_ticket(
+            subject="Billing issue",
+            description="I was charged twice",
+            customer_id=1,
```
Comment (yihua-zhuo) [inline tests/integration/test_ticket_categorization_integration.py:26]: **WARNING** (suggestion)

**Issue:** Integration tests hardcode customer_id=1 when creating tickets via TicketService, but no fixture or helper guarantees that customer ID 1 exists for the tenant under test. Tests may pass in isolation due to fixture ordering but are fragile.

**Suggestion:** Use a seed helper (e.g. _seed_customer) to explicitly create the customer first and use its returned ID, rather than relying on implicit fixture state or hardcoded IDs.

File: tests/unit/test_ticket_categorization_model.py [hunk 3]
```diff
@@ -0,0 +1,128 @@
+"""Unit tests for TicketCategorizationModel."""
```
Comment (yihua-zhuo) [inline tests/unit/test_ticket_categorization_model.py:1]: **WARNING** (suggestion)

**Issue:** Import uses bare 'db.models' instead of 'src.db.models', inconsistent with other test files in the same suite that use explicit 'src.' prefix (e.g. test_customer_model.py line 6).

**Suggestion:** Change 'from db.models.ticket_categorization import ...' to 'from src.db.models.ticket_categorization import ...'

File: tests/unit/test_ticket_categorization_model.py [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+        assert d["human_override"] is True
+        assert d["categorized_at"] == now.isoformat()
+
+    def test_to_dict_reasons_null_guard(self):
+        """reasons=None in model becomes {} in to_dict()."""
+        tc = TicketCategorizationModel(tenant_id=1, ticket_id=1, category_type="billing")
+        d = tc.to_dict()
+        assert d["reasons"] == {}
+
+    def test_to_dict_categorized_at_none(self):
```
Comment (yihua-zhuo) [inline tests/unit/test_ticket_categorization_model.py:106]: **WARNING** (suggestion)

**Issue:** Test assertion documents a potential bug rather than testing correct behavior: 'categorized_at=None in model becomes None string in to_dict()' should be None (Python None), not the string 'None'.

**Suggestion:** Rename test to reflect what it actually verifies (None stays None), or split into two tests: one asserting None input stays None, another asserting a set datetime serializes correctly.

**Comment by yihua-zhuo** [comment]
## Claude Code Review (inline)

**Status:** `pass`
**Inline comments posted:** 4

**Summary:** Reviewed 5 file(s) between origin/master and origin/auto/issue-602.

### Issues without a usable file/line
- `.plans/issue-602.md`: Plan document is well-structured and accurately describes the implementation scope, columns, indexes, test strategy, and acceptance criteria.
- `alembic/versions/a0000012_add_ticket_categorization.py`: Migration correctly defines all columns with proper types and constraints, uses SQL-side server defaults as required by project convention, creates the tenant_id index and composite index, and has a clean reversible downgrade.
- `tests/unit/test_ticket_categorization_model.py`: Test file covers initialization and to_dict() serialization but has no tests for model validation (invalid field types, out-of-range values), edge cases for the Decimal confidence field (e.g. > 1.0 or negative), or cross-tenant isolation behavior.
